### PR TITLE
key backup: Add a dedicated section to settings #251

### PR DESCRIPTION
**Pull request by BillCarsonFr**
### Pull Request Checklist

* [ ] Changes has been tested on an Android device or Android emulator with API 16
* [ ] Pull request is based on the develop branch
* [ ] Pull request updates [CHANGES.rst](https://github.com/vector-im/riot-android/blob/develop/CHANGES.rst)
* [ ] Pull request includes screenshots or videos if containing UI changes
* [ ] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.rst#sign-off)

### CODE REVIEW

File: vector/src/main/java/im/vector/fragments/keysbackup/settings/KeysBackupSettingsViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    private var _apiResultError: MutableLiveData<LiveEvent<String>> = MutableLiveData()
+    val apiResultError: LiveData<LiveEvent<String>>
+        get() = _apiResultError
+
+    var loadingEvent: MutableLiveData<LiveEvent<Int>> = MutableLiveData()
+
+    fun initSession(session: MXSession) {
+        this.session = session
+        keyBackupState.value = session.crypto?.keysBackup?.state
+        session.crypto?.keysBackup?.addListener(this)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/fragments/keysbackup/settings/KeysBackupSettingsViewModel.kt]: Only if this.session was null

File: vector/src/main/res/xml/vector_settings_preferences.xml [hunk 2]
```diff
@@ -334,6 +334,23 @@
...
             android:title="@string/encryption_information_device_key" />
 
+        <im.vector.preference.VectorSwitchPreference
+            android:key="SETTINGS_ENCRYPTION_NEVER_SENT_TO_PREFERENCE_KEY"
+            android:summary="@string/encryption_never_send_to_unverified_devices_summary"
+            android:title="@string/encryption_never_send_to_unverified_devices_title" />
+
+    </PreferenceCategory>
+
+    <im.vector.preference.VectorPreferenceDivider android:key="SETTINGS_CRYPTOGRAPHY_DIVIDER_PREFERENCE_KEY" />
```
Comment (bmarty) [inline vector/src/main/res/xml/vector_settings_preferences.xml]: Remove key if not used

File: vector/src/main/java/im/vector/fragments/keysbackup/settings/KeysBackupSettingsRecyclerViewAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+        val keyBackupState = session.crypto?.keysBackup?.state
+        val keyVersionResult = session.crypto?.keysBackup?.mKeysBackupVersion
+
+
+        val infos = ArrayList<GenericRecyclerViewItem>()
+        var itemSummary: GenericRecyclerViewItem? = null
+
+        keyBackupState?.let {
+            when (it) {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/fragments/keysbackup/settings/KeysBackupSettingsRecyclerViewAdapter.kt]: when(keyBackupState)

File: vector/src/main/java/im/vector/activity/KeysBackupManageActivity.kt [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        viewModel = ViewModelProviders.of(this).get(KeysBackupSettingsViewModel::class.java)
+        viewModel.initSession(mSession)
+
+
+        if (supportFragmentManager.fragments.isEmpty()) {
+            supportFragmentManager.beginTransaction()
+                    .replace(R.id.container, KeysBackupSettingsFragment.newInstance())
+                    .commitNow()
+
+            mSession.crypto?.keysBackup?.let {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/activity/KeysBackupManageActivity.kt]: Remove let

File: vector/src/main/res/xml/vector_settings_preferences.xml (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/res/xml/vector_settings_preferences.xml]: done

File: vector/src/main/java/im/vector/activity/KeysBackupManageActivity.kt (hunk 4 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/activity/KeysBackupManageActivity.kt]: done

**Pull request merged**